### PR TITLE
GH-1403: anneal against score and loss

### LINKED ISSUE #1403: When the dev is with a perfect score, consider a smaller loss an improvment
Hello,
I'm training a very simple dataset, and realized when it archive a perfect score, it do not consider the loss improvement as a good epoch. 

```
2020-02-03 04:45:49,242 EPOCH 42 done: loss 3.9541 - lr 0.0250
2020-02-03 04:46:11,877 DEV : loss 2.556217670440674 - score 1.0
2020-02-03 04:46:12,227 BAD EPOCHS (no improvement): 1
2020-02-03 04:46:18,934 ----------------------------------------------------------------------------------------------------
2020-02-03 04:46:20,809 epoch 43 - iter 1/15 - loss 0.11718750 - samples/sec: 0.55
2020-02-03 04:46:25,734 epoch 43 - iter 2/15 - loss 0.19140625 - samples/sec: 0.20
2020-02-03 04:46:26,519 epoch 43 - iter 3/15 - loss 0.13997396 - samples/sec: 1.32
2020-02-03 04:46:31,470 epoch 43 - iter 4/15 - loss 0.11376953 - samples/sec: 0.20
2020-02-03 04:46:35,815 epoch 43 - iter 5/15 - loss 0.12031250 - samples/sec: 0.23
2020-02-03 04:46:38,706 epoch 43 - iter 6/15 - loss 4.69742839 - samples/sec: 0.35
2020-02-03 04:46:41,963 epoch 43 - iter 7/15 - loss 4.03069196 - samples/sec: 0.31
2020-02-03 04:46:46,578 epoch 43 - iter 8/15 - loss 3.53417969 - samples/sec: 0.22
2020-02-03 04:46:48,116 epoch 43 - iter 9/15 - loss 3.14615885 - samples/sec: 0.66
2020-02-03 04:46:49,544 epoch 43 - iter 10/15 - loss 2.84052734 - samples/sec: 0.71
2020-02-03 04:46:51,052 epoch 43 - iter 11/15 - loss 2.58540483 - samples/sec: 0.67
2020-02-03 04:47:12,472 epoch 43 - iter 12/15 - loss 3.07373047 - samples/sec: 0.05
2020-02-03 04:47:33,290 epoch 43 - iter 13/15 - loss 2.84990986 - samples/sec: 0.05
2020-02-03 04:47:35,578 epoch 43 - iter 14/15 - loss 2.64885603 - samples/sec: 0.44
2020-02-03 04:47:37,917 epoch 43 - iter 15/15 - loss 2.50240885 - samples/sec: 0.43
2020-02-03 04:47:37,938 ----------------------------------------------------------------------------------------------------
2020-02-03 04:47:37,939 EPOCH 43 done: loss 2.5024 - lr 0.0250
2020-02-03 04:47:59,749 DEV : loss 1.2600260972976685 - score 1.0
2020-02-03 04:48:00,162 BAD EPOCHS (no improvement): 2
```

### PR DESCRIPTION
**Pull request by alanakbik**
Add new scheduler that uses dev score as main metric to anneal against, but additionally uses dev loss in case two epochs have the same dev score.

Closes #1403

### CODE REVIEW

File: flair/training_utils.py [hunk 1]
```diff
@@ -294,6 +297,172 @@ def _init_weights_index(self, key, state_dict, weights_to_watch):
...
+                is_better = True
+
+        if self.mode == 'max':
+            if current > self.best:
+                is_better = True
+
+        if current == self.best and auxiliary_metric:
+            current_aux = float(auxiliary_metric)
+            if self.aux_mode == 'min':
+                if current_aux < self.best_aux:
```
Comment (djstrong) [inline flair/training_utils.py:407]: Here 1self.best_aux1 has not been casted to float, but I do not know if it is necessary.

Comment (alanakbik) [reply flair/training_utils.py:407]: You're right - it's currently not necessary but consistency it would be better to cast as well.
